torch/_inductor/ir.py

```diff
@@ -3870,50 +3873,91 @@ def apply_constraint(self):
     def codegen_const_args(self):
         return map(V.graph.wrapper_code.val_to_arg_str, self.constant_args)
 
+    def collect_arg_kwarg_properties(self):
```

Comment:
It's still not clear to me where we should invoke ```collect_arg_kwarg_properties```. I see that we call it from multiple places like ```codegen_args``` and ```__init__``` functions. So, I guess my questions is - why is it not sufficient to just call it from ```__init__```? If it's not sufficient due to some side-effects happening between the ```__init__``` routine and ```codegen_args```, could we only call ```collect_arg_kwarg_properties``` from ```codegen_args``` instead? 

Unifying the call sites could help understand the code logic. Thanks!

Reply:
The main reason I chose to do it this way is because the base class, ExternKernel, is derived by many subclasses, so adding an extra field to the base class may require chances to all the subclass constructors. But you made a valid point. Let me update.